pixel/image.go

```diff
@@ -138,14 +138,14 @@ func (img Image[T]) setPixel(index int, c T) {
 	switch {
 	case zeroColor.BitsPerPixel() == 1:
 		// Monochrome.
-		x := index % int(img.width)
 		offset := index / 8
+		bits := index % 8
```

Comment:
The old behavior was most certainly incorrect.

Take for example an image of size 5x2. The first pixel at (0, 0) is `index` 0 and `x` 0 (make sense).
But at position (0, 1), which is index 5, you _also_ get `index` 0 and `x` 0. So the two pixels are stored at the same location.